### DISCUSSION TITLE
Array field ids

### DISCUSSION THREAD

**Discussion by yamila-fraiman** [react-hook-form/react-hook-form#9201] [General]
Hello, in the doc says that keyName will be removed in the next major version. How will it work? Using always id? Without ids?

Thank you.

**Comment by bluebill1049** [comment]
you won't need to change `keyName` prop anymore, field array will return id for each input group and not be part of your submission data.

**Reply by yamila-fraiman** [reply]
Thank you. Will it be part of fields array? Because I have my own id and actually I am using uuid as keyName to prevent overriding my mongodb id

**Reply by pfcodes** [reply]
Will it still override existing `id` properties in arrays?

**Comment by radeknowacki-toast** [comment]
@bluebill1049 I can see that in the latest version `keyName` should not be needed and IDs are handled automatically, but it doesn't seem to work for me. I've created a code sandbox to show the issue:

https://codesandbox.io/s/wispy-https-8y2uuo?file=/src/App.tsx:936-953

So if I don't specify `keyName` in FieldArray, my ids are overridden, but at the same time typescript doesn't seem to accept anything else than `id` as a value for `keyName`. Am I missing something?

EDIT: for anyone else wondering, it seems like bringing back original IDs happens in `handleSubmit`, which misled me. I am doing some transformations before sending my data to the API but I've done that too early when the id property stores form keys still.

**Reply by leapful** [reply]
@radeknowacki-toast 

Please use `getValues()` method or data object returned from on valid callback of `handleSubmit(onValid)` from `useForm` hook to get current form values.

The documentation of `fields` object from `useFieldArray` hook already described clearly that it only contains default value and key for React component, not the current field array value.

<img width="936" alt="Screenshot 2023-01-16 at 11 35 16" src="https://user-images.githubusercontent.com/18494222/212599490-2b3fe127-af6e-446a-a500-9f8730760dab.png">
